sdk/eventhub/Azure.Messaging.EventHubs/samples/Program.cs

```diff
@@ -42,27 +52,172 @@ public static async Task Main(string[] args)
+        ///   choice then it returns the chosen sample.
+        /// </summary>
+        ///
+        /// <returns>A sample or null if that could not be determined.</returns>
+        ///
+        private static INamedSample RetrieveSample()
+        {
+            var samples = LocateSamples();
+
+            PrintEventHubsSamples(samples);
+
+            var choice = ReadSelection(samples);
+
+            if (IsEventHubsIdentity(samples, choice))
+            {
+                var identitySamples = LocateIdentitySamples();
+
+                PrintEventHubsIdentitySamples(identitySamples);
+
+                while (IsEventHubsIdentity(samples, choice))
+                {
+                    choice = ReadSelection(identitySamples);
+
+                    if (choice.HasValue && !IsGoBack(identitySamples, choice))
+                    {
+                        return identitySamples[choice.Value];
+                    }
+                    else if (IsGoBack(identitySamples, choice))
+                    {
+                        choice = ReadSelection(samples);
+                    }
+                }
+            }
+
+            if (choice.HasValue)
+            {
+                return samples[choice.Value];
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        ///   It checks if an option is to go back.
+        /// </summary>
+        ///
+        /// <param name="samples">A list of identity samples</param>
+        /// <param name="choice">The choice made from console</param>
+        ///
+        /// <returns>If the user has chosed to go back in the main sample listing.</returns>
+        ///
+        private static bool IsGoBack(IReadOnlyList<IEventHubsIdentitySample> identitySamples, int? choice)
+        {
+            return choice == identitySamples.Count;
+        }
+
+        /// <summary>
+        ///   It checks if an option is to see the identity samples
+        /// </summary>
+        ///
+        /// <param name="samples">A list of samples</param>
+        /// <param name="choice">The choice made from console</param>
+        ///
+        /// <returns>If the user has chosen to see the event hubs identity samples.</returns>
+        ///
+        private static bool IsEventHubsIdentity(IReadOnlyList<IEventHubsSample> samples, int? choice)
+        {
+            return choice == samples.Count;
+        }
+
+        /// <summary>
+        ///   It prints to console a set of samples and controlling options
+        /// </summary>
+        ///
+        /// <param name="samples">A list of samples to be printed</param>
+        ///
+        private static void PrintEventHubsSamples(IReadOnlyList<INamedSample> samples)
+        {
+            PrintSamples(samples);
+
+            for (int i = 0; i < extraOptionsForSamples.Length; i++)
+            {
+                Console.WriteLine($"{ samples.Count + i + 1 }) {extraOptionsForSamples[i]}");
```

Comment:
nit:  If you would be so kind, would you put a space a between your `{` and the interpolated value?